app/src/main/java/br/com/concrete/tentacle/di/NetworkModule.kt

```diff
@@ -0,0 +1,47 @@
+package br.com.concrete.tentacle.di
+
+import br.com.concrete.tentacle.BuildConfig
+import br.com.concrete.tentacle.data.network.ApiService
+import com.google.gson.Gson
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import org.koin.dsl.module.module
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.concurrent.TimeUnit
+
+private const val CONNECTION_TIMEOUT = 15L
+private const val READ_TIMEOUT = 30L
+
+val networkModule = module {
+
+    single {
+        val httpLoggingInterceptor = HttpLoggingInterceptor()
+
+        if (BuildConfig.DEBUG) {
+            httpLoggingInterceptor.level = HttpLoggingInterceptor.Level.BODY
+        } else {
+            httpLoggingInterceptor.level = HttpLoggingInterceptor.Level.NONE
+        }
+
+        OkHttpClient.Builder()
+            .connectTimeout(CONNECTION_TIMEOUT, TimeUnit.SECONDS)
+            .readTimeout(READ_TIMEOUT, TimeUnit.SECONDS)
+            .addInterceptor(httpLoggingInterceptor).build()
+    }
+
+    single {
+        Retrofit.Builder()
+            .addConverterFactory(GsonConverterFactory.create(Gson()))
+            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+            .baseUrl(BuildConfig.BASE_URL)
```

Comment:
Uma pratica boa para receber a URL base do retrofit pensando em deixar o código testável e não ter que refazer o client do retrofit é receber essa variável por extraProperties, por exemplo. 

`startKoin(
            //Modules
            ),
            extraProperties = mapOf(PROPERTY_BASE_URL to BuildConfig.BASE_URL)
        )`

e no single do retrofit vocês podem pegar essa property da seguinte forma

`val baseUrl = getProperty<String>(PROPERTY_BASE_URL)`

Se quiserem dar uma olhada melhor nisso

https://insert-koin.io/docs/1.0/documentation/koin-android/index.html
https://insert-koin.io/docs/1.0/documentation/koin-core/index.html
https://insert-koin.io/docs/1.0/quick-references/koin-android/